### PR TITLE
tooling: add update-command script

### PR DESCRIPTION
**Pull request by vitorhcl**
- [X] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [X] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).

This script allows an update of the common part of a command on all languages. Its goal is reduce the need of manual work of translators to update pages when the descriptions and the placeholders don't change, so that they only need to review the changes.

### CODE REVIEW

**Comment by kbdharun** [comment]
In addition to @sebastiaanspeck's suggestion above, can you add the SPDX license identifier and some docs in the header? (feel free to refer to the other Python script files)

**Comment by vitorhcl** [comment]
> Could you run `black` on the Python-file?

I already runned it

**Comment by vitorhcl** [comment]
> Could you run `black` on the Python-file?

I already runned it

File: scripts/update-command.py
Comment (kbdharun) [inline scripts/update-command.py:1]:
```suggestion
from pathlib import Path
import os
```

File: scripts/update-command.py
Comment (vitorhcl) [reply scripts/update-command.py:24]: Done.

File: scripts/update-command.py
Comment (gutjuri) [inline scripts/update-command.py:8]:
```suggestion
    python3 scripts/update-command.py [-c] [-u] [-n] PLATFORM FILENAME [COMMAND]
```

File: scripts/update-command.py
Comment (kbdharun) [inline scripts/update-command.py:5]:
```suggestion
A Python script to update the common contents of a command across all languages.
```

File: scripts/update-command.py
Comment (kbdharun) [inline scripts/update-command.py:8]:
```suggestion
    python3 scripts/update-command.py [-c] [-u] [-n] <PLATFORM> <FILENAME> [COMMAND]
```

File: scripts/update-command.py
Comment (kbdharun) [inline scripts/update-command.py:24]:
```suggestion
       python3 scripts/update-command.py --dry-run -c "sudo apt install {{}}" -u "sudo apt install {{}} --no-confirm" linux apt
```

If not enclosed in proper quotes, the invoke the command modified too (or it exists before):

```zsh
(venv) ➜  tldr git:(tooling-update-command) ✗ python3 scripts/update-command.py --dry-run -c `sudo apt install {{}}` -u `sudo apt install {{}} --no-confirm` linux apt

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

E: Unable to locate package {{}}
E: Command line option --no-confirm is not understood in combination with the other options
usage: update-command.py [-h] [-c COMMON_PART] [-u UPDATED_COMMON_PART] [-n] [-v] platform filename
update-command.py: error: unrecognized arguments: Building dependency tree... Reading state information... apt
```

```zsh
(venv) ➜  tldr git:(tooling-update-command) ✗ python3 scripts/update-command.py --dry-run -c 'sudo apt install {{}}' -u 'sudo apt install {{}} --no-confirm' linux apt
Enter the commands (any content between double curly brackets will be ignored):
```

This is with double quotes:

```zsh
(venv) ➜  tldr git:(tooling-update-command) python3 scripts/update-command.py --dry-run -c "sudo apt install {{}}" -u "sudo apt install {{}} --no-confirm" linux apt
Enter the commands (any content between double curly brackets will be ignored):
`sudo apt install {{package}}` -> `sudo apt install {{package}} --no-confirm`
`sudo apt install {{paquet}}` -> `sudo apt install {{paquet}} --no-confirm`
`sudo apt install {{package}}` -> `sudo apt install {{package}} --no-confirm`
`sudo apt install {{pakket1 pakket2 ...}}` -> `sudo apt install {{pakket1 pakket2 ...}} --no-confirm`
`sudo apt install {{paket}}` -> `sudo apt install {{paket}} --no-confirm`
`sudo apt install {{الحزمة}}` -> `sudo apt install {{الحزمة}} --no-confirm`
`sudo apt install {{package}}` -> `sudo apt install {{package}} --no-confirm`
`sudo apt install {{pacchetto}}` -> `sudo apt install {{pacchetto}} --no-confirm`
`sudo apt install {{പാക്കേജ്}}` -> `sudo apt install {{പാക്കേജ്}} --no-confirm`
`sudo apt install {{パッケージ}}` -> `sudo apt install {{パッケージ}} --no-confirm`
`sudo apt install {{пакет}}` -> `sudo apt install {{пакет}} --no-confirm`
`sudo apt install {{nome_do_pacote}}` -> `sudo apt install {{nome_do_pacote}} --no-confirm`
`sudo apt install {{软件包}}` -> `sudo apt install {{软件包}} --no-confirm`
`sudo apt install {{paket}}` -> `sudo apt install {{paket}} --no-confirm`
`sudo apt install {{पैकेज}}` -> `sudo apt install {{पैकेज}} --no-confirm`
`sudo apt install {{প্যাকেজ}}` -> `sudo apt install {{প্যাকেজ}} --no-confirm`
`sudo apt install {{nome_do_pacote}}` -> `sudo apt install {{nome_do_pacote}} --no-confirm`
`sudo apt install {{paquete}}` -> `sudo apt install {{paquete}} --no-confirm`
`sudo apt install {{pakiet}}` -> `sudo apt install {{pakiet}} --no-confirm`
`sudo apt install {{நிரல்தொகுப்பு}}` -> `sudo apt install {{நிரல்தொகுப்பு}} --no-confirm`
`sudo apt install {{nama_paket}}` -> `sudo apt install {{nama_paket}} --no-confirm`
```

**Review by kbdharun — COMMENTED** [review]
Thanks for your contribution. I have some suggestions for the docs). Overall, the addition LGTM.

Can you add this to the README file in the `scripts` directory along with compatibility information? (It works fine on Linux and Windows; @sebastiaanspeck can you test it out on Mac?).

**Comment by kbdharun** [comment]
> The example states you can run the Python script from the tldr-root folder, but actually you need to be in the scripts folder:

The variable should still be the from tldr-root folder to be interoperable with the other scripts. Maybe similar to other scripts we can mention it here too.

> I also found out that it doesn't work well when trying to replace content with " or |.

Interesting, IG we can add disclaimer that states there could be some false positives so please review the change with the dry-run option before running it. I think this might have issues in pages where we escape placeholders too.

File: scripts/update-command.py
Comment (blueskyson) [inline scripts/update-command.py:144]: I think `place_placeholders` can be directly called rather than passed as `place_placeholders_function` argument. Is there any reason to do so?

File: scripts/update-command.py
Comment (blueskyson) [inline scripts/update-command.py:21]: There should be example inputs after users enter interactive mode.

File: scripts/update-command.py
Comment (blueskyson) [inline scripts/update-command.py:79]: `get_tldr_pages` and `is_placeholder` are never called in the script.

**Review by blueskyson — COMMENTED** [review]
I think this script is good to merge in terms of functionality. Other maintainers can open PR later if there's any concern. I'll approve after my comments are replied.

File: scripts/update-command.py
Comment (blueskyson) [inline scripts/update-command.py:100]: I think it's more precise to name the variable `commands` as `command_examples`.

**Review by gutjuri — APPROVED** [review]

**Review by kbdharun — APPROVED** [review]

**Review by blueskyson — APPROVED** [review]

**Pull request merged**
